examples/04_glm_first_level/plot_two_session_model.py

```diff
@@ -0,0 +1,247 @@
+)
+
+###############################################################################
+# Compute the statistics for the second session.
+fmri_glm_ses2 = fmri_glm.fit(fmri_imgs[1], design_matrices=design_matrices[1])
+
+# We will assume that the design matrix for the second session has the same
+# number of columns, in the same order, as the first session's.
+
+summary_statistics_ses2 = fmri_glm_ses2.compute_contrast(
+    contrast_val,
+    output_type='all',
+)
+plotting.plot_stat_map(
+    summary_statistics_ses2['z_score'],
+    bg_img=mean_img_,
+    threshold=3.0,
+    cut_coords=cut_coords,
+    title=f'{contrast_id}, second session',
+)
+
+###############################################################################
+# Compute the fixed effects statistics using both sessions' statistical maps.
+#
+# We can use :func:`~nilearn.glm.contrasts.compute_fixed_effects` to compute
+# the fixed effects statistics using the outputs from the session-specific
+# FirstLevelModel results.
+from nilearn.glm.contrasts import compute_fixed_effects
+
+contrast_imgs = [
+    summary_statistics_ses1['effect_size'],
+    summary_statistics_ses2['effect_size'],
+]
+variance_imgs = [
+    summary_statistics_ses1['effect_variance'],
+    summary_statistics_ses2['effect_variance'],
+]
+
+fixed_fx_contrast, fixed_fx_variance, fixed_fx_stat = compute_fixed_effects(
+    contrast_imgs,
+    variance_imgs,
+    data['mask'],
+)
+plotting.plot_stat_map(
+    fixed_fx_stat,
+    bg_img=mean_img_,
+    threshold=3.0,
+    cut_coords=cut_coords,
+    title=f'{contrast_id}, fixed effects',
+)
+
+###############################################################################
+# Not unexpectedly, the fixed effects version displays higher peaks than the
+# input sessions. Computing fixed effects enhances the signal-to-noise ratio of
+# the resulting brain maps.
+
+###############################################################################
+# Compute the fixed effects statistics using both sessions' preprocessed data.
+#
+# A more straightforward alternative to fitting session-specific GLMs, then
+# combining the results with
+# :func:`~nilearn.glm.contrasts.compute_fixed_effects`, is to simply fit the
+# GLM to both sessions at once.
+#
+# Since we can assume that the design matrices of both sessions have the same
+# columns, in the same order, we can again re-use the first session's contrast
+# vector.
+fmri_glm_multises = fmri_glm.fit(fmri_imgs, design_matrices=design_matrices)
+z_map = fmri_glm_multises.compute_contrast(
+    contrast_val,
+    output_type='z_score',
+)
+plotting.plot_stat_map(
+    z_map,
+    bg_img=mean_img_,
+    threshold=3.0,
+    cut_coords=cut_coords,
+    title=f'{contrast_id}, fixed effects',
+)
+
+plotting.show()
+
+###############################################################################
+# You may note that the results are the same as the first fixed effects
+# analysis, but with a lot less code.
```

Comment:
I said this, but the results are actually pretty different. I'm not sure why.

## compute_fixed_effects

![image](https://user-images.githubusercontent.com/8228902/159980430-1c952d20-a1e9-4401-b2e4-ecc25bce4814.png)


## GLM on both sessions

![image](https://user-images.githubusercontent.com/8228902/159980448-a3633058-e71c-462e-b674-15afb78a3572.png)

Reply:
It is because this is an F test, for which fixed effects are harder to define. I'm a bit worried by what you show.

Reply:
Ah, okay. Is this a matter of a bug existing in `compute_fixed_effects`, or is it only valid for t contrasts?

Reply:
Should be valid for F contrasts to, but I'm actually a bit worried given what you show. I'll have to double-check that.
Best

Reply:
Regarding the bug: it is actually in compute_fixed_effect(), that currently does not support F stats.
I think it is not really hard to make it work, but this will require a bit of work.
See #3194
For the time being, could we demo fixed effects on a t contrast ?

Reply:
Definitely! I'll switch to one of the t contrasts.

Reply:
Done in [4576c6e](https://github.com/nilearn/nilearn/pull/3191/commits/4576c6efc9bf5b3cfa6f4f4e8a0f78a3a8e1bfcb).

Reply:
There are still some minor differences in the two versions of the t contrast fixed effects maps, but at least they're a lot more similar than the F contrast ones were.

![image](https://user-images.githubusercontent.com/8228902/160200591-5fbdd8bd-9dfe-4d07-9581-672837483519.png)
![image](https://user-images.githubusercontent.com/8228902/160200599-a8d37178-56ad-4fbb-8a1b-80a7894105b0.png)

Reply:
Yes, there are minor differences which correspond to t stat/z stat differences.
This will be slightly improved in an upcoming PR, that I'm writing for the F test.